### PR TITLE
Adding .NET signatures to FDC3 documentation

### LINKED ISSUE #931: Proposal to add .NET interfaces and context declarations
### Use Case:
There is the need for .NET nuget packages for those that would like to implement FDC3 desktop agents or leverage other FDC3 capabilities in .NET applications.

For example: 
> An organization has a proprietary .NET desktop application and would like to use FDC3 API instead of proprietary or other mechanisms to communicate between .NET components.

> An organization has a proprietary .NET desktop container that hosts .NET and web based content via hosted browser (e.g. cefsharp/webview2).  There is a need to serialize/deserialize contexts between .NET and JavaScript when support exists for brokering messages between the two platforms.

> Consolidate .NET definitions to a single shared library for consistency with other .NET based FDC3 projects such as backplane.

### Additional Information
Morgan Stanley has published an alpha FDC3 v2 implementation that is ready for comment at https://github.com/morganstanley/fdc3-dotnet that can be contributed to FDC3.  This currently consists of two projects -

- [Main .NET standard library](https://github.com/morganstanley/fdc3-dotnet/tree/main/src/Fdc3) containing FDC3 interfaces, [context types](https://github.com/morganstanley/fdc3-dotnet/tree/main/src/Fdc3/Context), and some helper concrete implementations
- [Json .NET standard library](https://github.com/morganstanley/fdc3-dotnet/tree/main/src/Fdc3.Json) that has helper methods leveraging Newtonsoft.Json for any required custom serialization (e.g. Pascal case to camel case or union types such as Email recipients).

### PR DESCRIPTION
**Pull request by kriswest**
resolve #931 

Changes:
- Adds tabs throughout API spec for .NET signatures, 
- Moves supported platforms and API access content under API section

Reveiw notes: 
- Ensure a Changelog entry is added before merging
- Needs a vote of the SWG before merging

### CODE REVIEW

**Comment by kriswest** [comment]
@bingenito @robmoffat my meeting was cancelled so I used the time to get this started for. I've added tabs to every API signature and example with the content `TBC` in the .NET sections to be replaced with the proposed signatures. If we don't want to translate the examples we can always delete those tab blocks and retain the typescript examples only later (or encourage other contributors to add them).

To work on this, check out the FDC3 repo and switch to this branch. Then, in your terminal:

```bash
cd website
npm install
npm run start
```

That'll open a browser window with a preview from localhost. From there, select the 'next' version of the site from the version menu top right  - or you won't see any changes as it will default to displaying the current version (2.1). Or just hit this link: <http://localhost:3000/docs/next/api/spec>

The preview should update whenever you save - I suggest doing so often as you'll immediately see if you broke the page 😉 

I haven't yet done any prep in the Support Platforms page for .Net, but I did migrate all the JS-specific stuff there (which helpfully consolidated some duplicated content).

**Comment by kriswest** [comment]
@bingenito @robmoffat my meeting was cancelled so I got this out of the way. I've added tabs for .NET signatures and examples throughout the docs (although just noticed I missed some in the spec page, will fix). The .NET tabs all currently contain the content `TBC`. If you don't want to translate the examples we can always delete their tabs and stick with TS only, or we can encourage other contributors to take those on.

I've also migrated the JS/TS specific API access content under 'supported platforms' (clearing up some duplicated content along the way) and moved that page under the API section (as that's what it actually relates to) - I haven't prepped anything for .NET on that page as yet.

I also haven't touched the context pages as yet - we want to move to something generated from the schemas so I don't think that worth the effort - doing so will change the example from TypeScript to JSON. If we do something for .NET there I'd suggest we do it in the Context overview (as that will remain and won't be generated from schemas): <https://fdc3.finos.org/docs/next/context/spec>

To work on this, check out the branch then run the following in your terminal from the directory:

```bash
cd website
npm install
npm run start
```

That'll build it and open a browser window with a preview. Note you'll see the current version by default, to see your changes use the version menu at the top left to select the 'next' version of the spec (or use the local deep link below). The preview will update whenever you save - I suggest saving often as you'll find out if you broke the page 😉 

Local deep link: <http://localhost:3000/docs/next/api/ref/DesktopAgent>
Preview deep link: <https://deploy-preview-1108--fdc3.netlify.app/docs/next/api/spec>

**Comment by bingenito** [comment]
@kriswest Confirmed I can see your changes in local hosted site.  We will need to add support for csharp to the docusaurus config (it isn't on by default).

image

**Comment by bingenito** [comment]
Here we go.  I think you can close this PR if you want and we can continue working on this branch in tandem to create eventual PR.

image

**Comment by kriswest** [comment]
@bingenito I converted this to a draft PR for now - I find the diffs and preview they provide useful and it gives us a thread to discuss as we go ;-) 

Did you push your change to the docusaurus config?

**Comment by kriswest** [comment]
>@kriswest I don't have permission to push to this branch. Do you want me to do my contribution through a PR to this branch and then we PR to master for group review?

@robmoffat @maoo now that we've got branch protection rules in place for the master branch of the FDC3 repo, is it possible to give push rights to the fdc3-participants team (which Brian is in) to facilitate collaboration by participants on PRs like this? 

If not, do we need to use a fork that we both have rights to achieve the same thing elsewhere and/or PR into this branch from a fork? Please advise on best practice :-)
The final alternative is that we get Brian appointed as a maintainer (which we propose to do anyway with the .NET contribution) - although I'm sure this will come up again with others (and we'll follow whatever procedure you advise/require).

**Comment by maoo** [comment]
Hi @kriswest !

The general best practice is to fork, invite, collaborate and submit PR, as both you and @bingenito suggested.

That said, given that Brian wrote the .net client, and is willing to collaborate and maintain this part of code, I believe he's a candidate to become an FDC3 maintainer. As FDC3 lead maintainer, you have the authority (and rights, via https://github.com/orgs/finos/teams/fdc3-maintainers) to make it happen, assuming other maintainers are ok with it.

**Comment by bingenito** [comment]
@kriswest I added a very simple native/.net section to get something out the door.  I had a little incident with origin vs upstream that is showing as a strange merge commit in the PR.  The diff looks on that one but I'm not sure what git did - whether it reverted a change you made.

**Comment by kriswest** [comment]
Added a changelog entry and changed this PR to ready to review as I think it is @bingenito. 

It'll need a review from @hughtroeger and perhaps another member of @finos/fdc3-maintainers.

**Comment by kriswest** [comment]
@bingenito you were a contributor to this PR but are currently not the last push if you want to add a review yourself.

**Review by bingenito — DISMISSED** [review]

**Comment by kriswest** [comment]
Thanks @bingenito - we just need @hughtroeger's blessing in that case ;-)

**Review by bingenito — APPROVED** [review]

**Review by bingenito — APPROVED** [review]

**Comment by bingenito** [comment]
@kriswest Should globals.md be showing as deleted?

**Comment by bingenito** [comment]
> @kriswest Should globals.md be showing as deleted?
Looks like it was in the very first commit with the creation of supported platforms.  I guess I had just glossed over it being deleted.

**Comment by kriswest** [comment]
Yes, it's only relevant to web so it made sense to move to supported-platforms. fdc3-for-the-web will replace the recommendation with using getAgent() anyway, so its reworked further in that PR.

**Review by hughtroeger — APPROVED** [review]

File: docs/api/ref/Errors.md
Comment (kriswest) [reply docs/api/ref/Errors.md:50]: I think thats all the formatting they get in prism - as a comment block.

**Comment by kriswest** [comment]
🎉 we (well @bingenito) got there! Merging...

File: docs/api/ref/Errors.md
Comment (bingenito) [reply docs/api/ref/Errors.md:50]: c# doc syntax for intellisense and if you were to do doc generation.

**Pull request merged**
